antarest/matrixstore/service.py

```diff
@@ -0,0 +1,88 @@
+import time
+from datetime import datetime
+from typing import List, Optional, Tuple
+
+from antarest.common.custom_types import JSON
+from antarest.matrixstore.model import (
+    MatrixDTO,
+    MatrixFreq,
+    MatrixType,
+    Matrix,
+    MatrixContent,
+)
+from antarest.matrixstore.repository import (
+    MatrixRepository,
+    MatrixContentRepository,
+)
+
+
+class MatrixService:
+    def __init__(
+        self, repo: MatrixRepository, content: MatrixContentRepository
+    ):
+        self.repo = repo
+        self.repo_content = content
+
+    @staticmethod
+    def _to_dto(matrix: Matrix, content: MatrixContent) -> MatrixDTO:
+        return MatrixDTO(
+            id=matrix.id,
+            type=matrix.type,
+            freq=matrix.freq,
+            created_at=int(time.mktime(datetime.timetuple(matrix.created_at))),
+            updated_at=int(time.mktime(datetime.timetuple(matrix.updated_at))),
+            index=content.index,
+            columns=content.columns,
+            data=content.data,
+        )
+
+    @staticmethod
+    def _from_dto(dto: MatrixDTO) -> Tuple[Matrix, MatrixContent]:
+        matrix = Matrix(
+            id=dto.id,
+            type=dto.type,
+            freq=dto.freq,
+            created_at=datetime.fromtimestamp(dto.created_at),
+            updated_at=datetime.fromtimestamp(dto.updated_at),
+        )
+
+        content = MatrixContent(
+            data=dto.data, index=dto.index, columns=dto.columns
+        )
+
+        return matrix, content
+
+    def save(self, data: MatrixDTO) -> str:
```

Comment:
Indeed matrix is immutable. So we can only create matrix. Ok for refactor. But since we share matrix, service doesn't care if matrix already exist or not

Reply:
oh.. I thought the matrix would be mutable, which indeed would be better not. So you can remove the update_at field. 'save' is ok since there won't be an 'update'